packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+import { mockConfigProvider } from "./mockConfigProvider";
+
+class TestNode {
+    public readonly parents: Set<string> = new Set();
+    public readonly children: Set<string> = new Set();
+    constructor(public readonly id: string) {}
+}
+
+class IgnoreErrorLogger extends EventAndErrorTrackingLogger {
+    constructor(protected readonly baseLogger: ITelemetryBaseLogger) {
+        super(baseLogger);
+    }
+
+    private readonly ignoredEvents: Map<string, ITelemetryGenericEvent> = new Map();
+
+    public ignoreExpectedEventTypes(... anyIgnoredEvents: ITelemetryGenericEvent[]) {
+        for (const event of anyIgnoredEvents) {
+            this.ignoredEvents.set(event.eventName, event);
+        }
+    }
+
+    send(event: ITelemetryBaseEvent): void {
+        if (this.ignoredEvents.has(event.eventName)) {
+            let matches = true;
+            const ie = this.ignoredEvents.get(event.eventName);
+            assert(ie !== undefined);
+            for (const key of Object.keys(ie)) {
+                if (ie[key] !== event[key]) {
+                    matches = false;
+                    break;
+                }
+            }
+            if (matches) {
+                event.category = "generic";
+            }
+        }
+
+        this.baseLogger.send(event);
+    }
+}
+
+/**
+ * Validates this scenario: When a datastore is aliased that it is considered a root datastore and always referenced
+ */
+describeNoCompat("GC Random tests", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            summaryConfigOverrides: {
+                state: "disableHeuristics",
+                initialSummarizerDelayMs: 0,
+                summarizerClientElection: false,
+                maxAckWaitTime: 100,
+                maxOpsSinceLastSummary: 100,
+            },
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const defaultSessionExpiryDurationMs = 3000; // 3 seconds
+
+    // Enable config provider setting to write GC data at the root.
+    const settings = {
+        "Fluid.GarbageCollection.RunSessionExpiry": "true",
+        "Fluid.GarbageCollection.TestOverride.SessionExpiryMs": defaultSessionExpiryDurationMs,
+    };
+    const configProvider = mockConfigProvider(settings);
```

Comment:
Oh, this looks much nicer than clobbering the sessionStorageConfigProvider like Navin and I have been doing